vm/vmStmt.go

```diff
@@ -345,18 +364,27 @@ func runSingleStmt(stmt ast.Stmt, env *Env) (reflect.Value, error) {
 			newenv := env.NewEnv()
 
 			for {
-				if *(env.interrupt) {
+				cases := []reflect.SelectCase{{
+					Dir:  reflect.SelectRecv,
+					Chan: reflect.ValueOf(ctx.Done()),
+					Send: reflect.ValueOf(nil),
+				}, {
+					Dir:  reflect.SelectRecv,
+					Chan: val,
+					Send: reflect.ValueOf(nil),
+				}}
+				chosen, iv, ok := reflect.Select(cases)
+				if chosen == 0 {
```

Comment:
Noticed this one is not being covered by the tests. Not required but would be nice if it could be covered by the tests.

Reply:
Yeah, I noticed it as well. Couldn't figure out out to reach this part of the software :/

Reply:
Updated.